### PR TITLE
rustic-rs: init at 0.3.2

### PR DESCRIPTION
**Pull request by NobbZ**
###### Description of changes

fast, encrypted, deduplicated backups powered by pure Rust

https://github.com/rustic-rs/rustic (Currently no dedicated homepage)

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
Can you also install shell completions like here? https://github.com/NixOS/nixpkgs/blob/013fcdd106823416918004bb684c3c186d3c460f/pkgs/applications/science/machine-learning/finalfrontier/default.nix#L39
There is a `completions` subcommand.

**Review by Atemu — CHANGES_REQUESTED** [review]
Result of `nixpkgs-review pr 201135` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>rustic-rs</li>
  </ul>
</details>

**Comment by Mic92** [comment]
Blind fix for the macOS issue:

```
    buildInputs = lib.optionals stdenv.isDarwin [ darwin.Security ];
```

**Comment by Atemu** [comment]
You usually reference the individual deps as inputs and inherit them from `darwin` in `all-packages`'s `callPackage`.

@NobbZ do you have access to a Darwin machine? Otherwise limit the support to Linux and someone else can add support for Darwin later.

File: pkgs/tools/backup/rustic-rs/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{lib, fetchFromGitHub, rustPlatform}:
```
[the remark this answers is not in the document]
Comment (NobbZ) [reply pkgs/tools/backup/rustic-rs/default.nix:1]: Oh yeah :D I'm used to much to the alejandra style and haven't seen this up here… I remembered the licenses at least :D

**Comment by NobbZ** [comment]
Yeah, I have some access, and hope to find some time to play with it later today. I need to fix the nix installation first though.

I will draft this PR for now.

**Comment by NobbZ** [comment]
Damn GH not allowing to change a PRs state to draft after it has been created…

I hope the title makes it obvious enough for the start…

**Comment by NobbZ** [comment]
Thanks for the draft @Mic92! Is that a Commiters superpower or did I just miss a button somewhere?

Anyway, I was able to fix it on the mac, now am crosschecking on my x86 system.

**Comment by NobbZ** [comment]
Yupp worked there.

I think it is ready for review then.

**Comment by Mic92** [comment]
> Thanks for the draft @Mic92! Is that a Commiters superpower or did I just miss a button somewhere?

I don't think so. It is usually below the Reviewers box at the right side.

**Comment by NobbZ** [comment]
> I don't think so. It is usually below the Reviewers box at the right side.

I will keep the eyes open on my next PR :D

Thanks for the merge!

File: pkgs/tools/backup/rustic-rs/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ lib, fetchFromGitHub, rustPlatform, stdenv, darwin, installShellFiles }:
```
Comment (Atemu) [inline pkgs/tools/backup/rustic-rs/default.nix:1]: As mentioned, elsewhere in Nixpkgs it's usually done like this

```Suggestion
{ lib, fetchFromGitHub, rustPlatform, stdenv, Security, installShellFiles }:
```

an then explicitly inherited from `darwin` at `callPackage`.

I have no idea why that is (I'd much rather the frameworks were available as regular attrs for `callPackage` magic) but note that this goes against the currently practiced convention.

Comment (NobbZ) [reply pkgs/tools/backup/rustic-rs/default.nix:1]: When you said to "inherit darwin from callpackage" I though you mean it exactly like this.

I will take a look into how it is done elsewhere and then try to remember to do it that way on the next bump.

Comment (Mic92) [reply pkgs/tools/backup/rustic-rs/default.nix:1]: I am aware fo the pattern but I don't get why its done. Do people really override frameworks individually? Is this even possible? To me it still looks like cargo culting.

Comment (Atemu) [reply pkgs/tools/backup/rustic-rs/default.nix:1]: As I said, I have no idea why it's done. Just noting that it's the current convention.

**Pull request merged**